lfs.c

```diff
@@ -4404,23 +4430,16 @@ static int lfs_mount_(lfs_t *lfs, const struct lfs_config *cfg) {
 
     // scan directory blocks for superblock and any global updates
     lfs_mdir_t dir = {.tail = {0, 1}};
-    lfs_block_t tortoise[2] = {LFS_BLOCK_NULL, LFS_BLOCK_NULL};
-    lfs_size_t tortoise_i = 1;
-    lfs_size_t tortoise_period = 1;
+    struct lfs_tortoise_t tortoise = {
+        .pair = {LFS_BLOCK_NULL, LFS_BLOCK_NULL},
+        .i = 1,
+        .period = 1,
+    };
     while (!lfs_pair_isnull(dir.tail)) {
-        // detect cycles with Brent's algorithm
-        if (lfs_pair_issync(dir.tail, tortoise)) {
-            LFS_WARN("Cycle detected in tail list");
-            err = LFS_ERR_CORRUPT;
+        err = lfs_detect_cycles(&dir, &tortoise);
+        if (err == LFS_ERR_CORRUPT) {
```

Comment:
We probably want:
``` c
if (err) {
    goto cleanup;
}
```

Otherwise we miss any other errors returned by `lfs_detect_cycles`. It's true `lfs_detect_cycles` doesn't return anything but LFS_ERR_CORRUPT, but these sort of assumptions can cause problems for future refactoring.